docs/docs/dev_docs/privacy/main.md

```diff
@@ -0,0 +1,165 @@
+  - All arguments which are passed to public functions.
+- All calls to L1 functions (in the form of L2 -> L1 messages).
+  - The contents of L2 -> L1 messages.
+- All unencrypted logs (topics and arguments).
+- The roots of all trees which have been read from.
+- The _number_ of ['side effects'](https://en.wikipedia.org/wiki/Side_effect_(computer_science)):
+  - \# new commitments
+  - \# new nullifiers
+  - \# bytes of encrypted logs
+  - \# public function calls
+  - \# L2->L1 messages
+  - \# nonzero roots**
+
+The leakiness of a Tx Fingerprint depends on the leakiness of its consituent functions' Function Fingerprints _and_ on the appearance of the tx's Tx Fingerprint as a whole. Txs can leak the following similar information:
+- All calls to public functions.
+  - The contract addresses of all private function (if it calls an internal public function).
+    - This could be the address of the transactor themselves, if the calling contract is an account contract.
+  - All arguments which are passed to public functions.
+- All calls to L1 functions (in the form of L2 -> L1 messages).
+  - The contents of L2 -> L1 messages.
+- All unencrypted logs (topics and arguments).
+- The roots of all trees which have been read from.
+- The _number_ of ['side effects'](https://en.wikipedia.org/wiki/Side_effect_(computer_science)) in the whole tx:
+  - \# new commitments
+  - \# new nullifiers
+  - \# bytes of encrypted logs
+  - \# public function calls
+  - \# L2->L1 messages
+  - \# nonzero roots[^1]
+
+
+> Note: many of these were mentioned in the ["Crossing the private -> public boundary"](#crossing-the-private---public-boundary) section.
+
+> Note: the calldata submitted to L1 is [encoded](https://github.com/AztecProtocol/aztec-packages/blob/master/l1-contracts/src/core/libraries/Decoder.sol) in such a way that all categories of data are packed together, when submitted. E.g. all commitments from all txs in a block are arranged as contiguous bytes of calldata. But that _doesn't_ mean the data from a particular tx is garbled in with all other txs' calldata: the distinct Tx Fingerprint of each tx can is publicly visible when a tx is submitted to the L2 tx pool.
+
+
+#### Standardising Fingerprints
+
+If each private function were to have a unique Fingerprint, then all private functions would be distinguishable from each-other, and all of the efforts of the Aztec protocol to enable 'private function execution' would have been pointless. Standards need to be developed, to encourage smart contract developers to adhere to a restricted set of Tx Fingerprints. For example, a standard might propose that the number of new commitments, nullifiers, logs, etc. must always be equal, and must always equal a power of two. Such a standard would effectively group private functions/txs into 'privacy sets', where all functions/txs in a particular 'privacy set' would look indistinguishable from each-other, when executed.
+
+
+### Data queries
```

Comment:
Much of this seems duplicaty of what was in limitations.

Reply:
Yeah, I think that's ok to have some overlap here. In the limitations sections it's a very brief outline, whereas this goes into more detail.